scripts/snapshot-calculator.ts

```diff
@@ -0,0 +1,161 @@
+import { config } from "dotenv";
+import { ethers } from "hardhat";
+import { writeFileSync } from "node:fs";
+import { Log, JsonRpcProvider } from "@ethersproject/providers";
+import { Contract, Wallet, providers, utils } from "ethers";
+import { Snapshot } from "./types/snapshot.types";
+import { EW_CHAIN_ID, formatDID, _rpcReadContractSlot, getSlotNumber } from "./utils/snapshot.utils";
+
+config();
+let stakingContract: Contract;
+let failingCalculations: string[] = [];
+
+const STAKES_STORAGE_SLOT = getSlotNumber("StakingPoolPatronKYC", "stakes"); //slot num 12
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const nodeEmoji = require("node-emoji");
+const wallet = new Wallet("9d69bf3c7bda0fc771a59818e5a7f894f3b4fbd3114891db3d4de7f46aa71ca8");
+
+const parseEvents = async (stakeLogs: Log[], blockNumber: number) => {
+  const stakers: string[] = [];
+  await Promise.all(
+    stakeLogs.map(async (currentLog) => {
+      if (currentLog.blockNumber <= blockNumber) {
+        //Removing the padding zeros
+        const currentAddress = utils.hexStripZeros(currentLog.topics[1]);
+        if (!stakers.includes(currentAddress)) {
+          stakers.push(currentAddress);
+        }
```

Comment:
Similar to my other comment, better to use a `Set` here
```suggestion
        stakers.add(currentAddress)
```